clang/lib/Sema/SemaCUDA.cpp

```diff
@@ -115,20 +143,65 @@ static bool hasAttr(const Decl *D, bool IgnoreImplicitAttr) {
          });
 }
 
+SemaCUDA::CUDATargetContext::CUDATargetContext(SemaCUDA *S,
```

Comment:
This could probably be moved into the header.

Reply:
Addressed in e273a996854cb3256cfeed832c3f16ff6fd0f534.